source/agora/consensus/Ledger.d

```diff
@@ -535,6 +535,7 @@ public class Ledger
                 log.fatal("Validated block: {}", block);
                 assert(0);
             }
+            this.utxo_set.updateUTXOLock(enrollment.utxo_key, block.header.height + this.params.ValidatorCycle);
```

Comment:
we have to lock the UTXO for the enrollment duration. if not, validator can spend the stake and the penalty deposit. Even tho the outputs are locked (ie melting), we no longer can slash them since we lose track of the stake.